inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp

```diff
@@ -738,714 +738,366 @@ struct typed_resizeLinearU8C3C4 {
 
-        rowFunc(in.InLineB(0), chan, in.meta().chan, out.OutLineB(), in.length());
-    }
-};
+    inline Unit map(int outCoord) {
+        double inCoord0 =  outCoord      * ratio;
+        double inCoord1 = (outCoord + 1) * ratio;
 
-GAPI_FLUID_KERNEL(FNV12toRGB, NV12toRGB, false) {
-    static const int Window = 1;
-    static const int LPI = 2;
-    static const auto Kind = cv::GFluidKernel::Kind::YUV420toRGB;
+        double index0 = std::floor(inCoord0 + 0.001);
+        double index1 =  std::ceil(inCoord1 - 0.001);
 
-    static void run(const cv::gapi::fluid::View & in_y,
-                    const cv::gapi::fluid::View & in_uv,
-                    cv::gapi::fluid::Buffer & out) {
-        GAPI_DbgAssert(is_cv_type_in_list<nv12_to_rgb_supported_types>(out.meta().depth));
+        double alpha0 =   (index0 + 1 - inCoord0) * inv_ratio;
+        double alpha1 = - (index1 - 1 - inCoord1) * inv_ratio;
 
-        const uchar* uv_row = in_uv.InLineB(0);
-        const uchar* y_rows[2] = { in_y.InLineB(0), in_y.InLineB(1) };
-        uchar* out_rows[2] = { out.OutLineB(0), out.OutLineB(1) };
+        GAPI_Assert(0 <= outCoord && outCoord <= outSz-1);
+        GAPI_Assert(0 <= index0 && index0 < index1 && index1 <= inSz);
 
-        int buf_width = out.length();
+        Unit unit;
 
-        const auto rowFunc = type_dispatch<nv12_to_rgb_supported_types>(out.meta().depth, cv_type_id{}, typed_nv12_to_rgb_row<isa_tag_t>{}, nullptr);
+        unit.index0 = checked_cast<index_type>(index0);
+        unit.index1 = checked_cast<index_type>(index1);
 
-        GAPI_DbgAssert(rowFunc);
+        unit.alpha0 = convert_cast<alpha_type>(alpha0);
+        unit.alpha1 = convert_cast<alpha_type>(alpha1);
 
-        rowFunc(y_rows, uv_row, out_rows, buf_width);
+        return unit;
     }
-};
-
-GAPI_FLUID_KERNEL(FI420toRGB, I420toRGB, false) {
-    static const int Window = 1;
-    static const int LPI = 2;
-    static const auto Kind = cv::GFluidKernel::Kind::YUV420toRGB;
-
-    static void run(const cv::gapi::fluid::View & in_y,
-                    const cv::gapi::fluid::View & in_u,
-                    const cv::gapi::fluid::View & in_v,
-                    cv::gapi::fluid::Buffer & out) {
-        GAPI_DbgAssert(is_cv_type_in_list<i420_to_rgb_supported_types>(out.meta().depth));
 
-        const uchar* u_row = in_u.InLineB(0);
-        const uchar* v_row = in_v.InLineB(0);
-        const uchar* y_rows[2] = { in_y.InLineB(0), in_y.InLineB(1) };
-        uchar* out_rows[2] = { out.OutLineB(0), out.OutLineB(1) };
+    int    inSz, outSz;
+    double ratio, inv_ratio;
 
-        int buf_width = out.length();
-        GAPI_DbgAssert(in_u.length() == in_v.length());
+    alpha_type  alpha;  // == inv_ratio, rounded
 
-        const auto rowFunc = type_dispatch<i420_to_rgb_supported_types>(out.meta().depth, cv_type_id{}, typed_i420_to_rgb_row<isa_tag_t>{}, nullptr);
+    void init(int _inSz, int _outSz) {
```

Comment:
Could it be a part of constructor? to keep object consistent straight after creation

Reply:
Done.